examples/src/streaming/server.rs

```diff
@@ -29,28 +56,37 @@ impl pb::echo_server::Echo for EchoServer {
         &self,
         req: Request<EchoRequest>,
     ) -> EchoResult<Self::ServerStreamingEchoStream> {
-        println!("Client connected from: {:?}", req.remote_addr());
+        println!("EchoServer::server_streaming_echo");
+        println!("\tclient connected from: {:?}", req.remote_addr());
 
-        let (tx, rx) = oneshot::channel::<()>();
-
-        tokio::spawn(async move {
-            let _ = rx.await;
-            println!("The rx resolved therefore the client disconnected!");
+        // creating infinite stream with requested message
+        let repeat = std::iter::repeat(EchoResponse {
+            message: req.into_inner().message,
         });
+        let mut stream = Box::pin(tokio_stream::iter(repeat).throttle(Duration::from_millis(200)));
 
-        struct ClientDisconnect(oneshot::Sender<()>);
-
-        impl Stream for ClientDisconnect {
-            type Item = Result<EchoResponse, Status>;
-
-            fn poll_next(self: Pin<&mut Self>, _: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-                // A stream that never resolves to anything....
-                Poll::Pending
+        // spawn and channel is required if you want handle "disconnected" functionality
+        let (tx, rx) = mpsc::channel(1);
+        tokio::spawn(async move {
+            while let Some(item) = stream.next().await {
+                match tx.send(Result::<_, Status>::Ok(item)).await {
+                    Ok(_) => {
+                        // the item was send to channel but there is no guarntee it was send to client!
+                        // If you need such guarntee the solution can be "rendezvous channel"
+                        // (channel with 0 capacity), tokio::sync::mpsc do not support this special case
+                        // but you can use `flume::bounded` https://docs.rs/flume/latest/flume/fn.bounded.html
+                    }
+                    Err(_item) => {
+                        println!("\tclient disconnected"); //out_stream (build on rx) was droped
+                        break;
+                    }
+                }
```

Comment:
So I kinda disagree with this method of tracking client disconnects. I would 1) I don't want to suggest flume for a few reason I don't want to dig into right now but at a high level we shouldn't be recommending it to users. 2) If you want to know if your client got a message this should be handled at a high level at the application level via acks. This is because no matter what with tcp/udp there is no way to ensure a user actually got the message. So the best way is to get an ack on the user level code that the users client has gotten the message. 

If a client disconnects the `while let` loop should exit.